libcloudforensics/providers/gcp/internal/gke.py

```diff
@@ -36,24 +41,105 @@ def GkeApi(self) -> 'googleapiclient.discovery.Resource':
     Returns:
       googleapiclient.discovery.Resource: A Google Container service object.
     """
+    return common.CreateService('container', self.GKE_API_VERSION)
 
-    return common.CreateService(
-        'container', self.GKE_API_VERSION)
 
-  def GetCluster(self, name: str) -> Dict[str, Any]:
-    """ Gets the details of a specific cluster.
+class GkeCluster(GoogleKubernetesEngine):
+  """Class to call GKE and Kubernetes APIs on a GKE resource.
+
+  https://cloud.google.com/kubernetes-engine/docs/reference/rest
+  https://kubernetes.io/docs/reference/
+
+  Attributes:
+    project_id (str): The GCP project name.
+    zone (str): The GCP zone for this project.
+    cluster_id (str): The name of the GKE cluster.
+  """
+
+  def __init__(self, project_id: str, zone: str, cluster_id: str) -> None:
+    """Creates a GKE cluster resource.
 
     Args:
-      name (str): The name (project, location, cluster) of the cluster to
-          retrieve. Specified in the format `projects/*/locations/*/clusters/*`.
-          For regional cluster: `/locations/[GCP_REGION]`.
-          For zonal cluster: `/locations/[GCP_ZONE]`.
+      project_id (str): The GCP project name.
+      zone (str): The GCP zone for this project.
+      cluster_id (str): The name of the GKE cluster.
+    """
+    self.project_id = project_id
+    self.zone = zone
+    self.cluster_id = cluster_id
+
+  @property
+  def name(self) -> str:
+    """Name of the GKE cluster resource, for use in API calls.
+
+    Returns:
+      str: Full name of the cluster resource.
+    """
+    return 'projects/{0:s}/locations/{1:s}/clusters/{2:s}'.format(
+      self.project_id,
+      self.zone,
+      self.cluster_id,
+    )
+
+  def GetCredentials(self) -> client.ApiClient:
```

Comment:
GetCredentials name might not be the most suitable here, maybe something that refers to an authenticated API client would be better, something with similar meaning to GetAuthenticatedK8sApiClient